poc/map/Jack.tbasic

```diff
@@ -0,0 +1,778 @@
+        me.worldList(i).arms(j).depthProfile = getRandomFlag(me.worldList(i).ambiance.armsDepthProfiles)
+      Next
+        
+    	Case %MOSTLYNARROW
+    	
+        narrowCount = 0
+        narrowCurrent = 3
+        narrowTotal = 0
+        
+        for j as dword = 1 to me.worldList(i).ambiance.riverArmsCount
+          incr narrowCount
+          if narrowCount > 3 then
+          	narrowCount = 0
+          	narrowCurrent += 2
+          End If
+          'Width
+          if j = me.worldList(i).ambiance.riverArmsCount Then
+            me.worldList(i).arms(j).width += iseven(assign2(me.worldList(i).minimumTotalRiverWidth - narrowTotal, varptr(me.worldList(i).arms(j).width)))
+          else
+            me.worldList(i).arms(j).width = narrowCurrent
+            narrowTotal += narrowCurrent
+          endif
+          
+          'me.worldList(i).widestArmWidth = max(me.worldList(i).widestArmWidth, me.worldList(i).arms(j).width)
+          
+          'Random depth profile
+          me.worldList(i).arms(j).depthProfile = getRandomFlag(me.worldList(i).ambiance.armsDepthProfiles)
+        Next
+        
+    	Case %MOSTLYWIDE
+    	
+        wideTotal = 0
+        
+        for j as dword = 1 to me.worldList(i).ambiance.riverArmsCount
+          'Width
+          if j = me.worldList(i).ambiance.riverArmsCount Then
+            me.worldList(i).arms(j).width += iseven(assign2(me.worldList(i).minimumTotalRiverWidth - wideTotal, varptr(me.worldList(i).arms(j).width)))
+          Else
+            me.worldList(i).arms(j).width += iseven(assign2(ceil(me.worldList(i).minimumTotalRiverWidth / (me.worldList(i).ambiance.riverArmsCount + j - 1)), varptr(me.worldList(i).arms(j).width)))
+            wideTotal += me.worldList(i).arms(j).width
+          endif
+          
+          'me.worldList(i).widestArmWidth = max(me.worldList(i).widestArmWidth, me.worldList(i).arms(j).width)
+          
+          'Random depth profile
+          me.worldList(i).arms(j).depthProfile = getRandomFlag(me.worldList(i).ambiance.armsDepthProfiles)
+        Next
+        
+    End Select
+    
+
+    '[todo] <Something to do here>
+    ' Here sorting me.worldList(i).arms(j).width wide to narrow ; how array sort ?
+   #Region " Sort arms by width descending"
+    Dim sorted as Boolean
+    DIm tempArm as tArm
+      '[breakpoint] <Any note here. Breakpoint will be set to line following this statement>
+    Do
+      sorted = %TRUE
+      if me.worldList(i).ambiance.riverArmsCount = 1 then exit do
+      for j = 2 to me.worldList(i).ambiance.riverArmsCount ' should be " AS integer " but I have a bug there
+        if me.worldList(i).arms(j).width > me.worldList(i).arms(j-1).width Then
+        	sorted = %False
+        	tempArm = me.worldList(i).arms(j).width
+        	me.worldList(i).arms(j-1).width = me.worldList(i).arms(j).width
+        	me.worldList(i).arms(j).width = tempArm
+        End If
+      Next
+    loop until sorted = %TRUE
+   #EndRegion
+donothing()
```

Comment:
If you comment out line 462 then the script won't finish. It's beyond all (my) comprehension.